src/Zicht/Bundle/SolrBundle/Solr/Client.php

```diff
@@ -83,19 +87,29 @@ protected function doRequest(QueryBuilder\RequestBuilderInterface $handler)
             }
 
             $this->logs[] = ['response' => $response, 'requestUri' => $request->getUri()];
-        } catch (BadResponseException $e) {
-            $this->lastResponse = $e->getResponse();
-            if ($e->getRequest()->getBody()) {
-                $e->getRequest()->getBody()->seek(0);
+        } catch (BadResponseException $be) {
+            $this->lastResponse = $be->getResponse();
+            if ($be->getRequest()->getBody()) {
+                $be->getRequest()->getBody()->seek(0);
+            }
+            $content = $be->getResponse()->getBody()->getContents();
+            $contentType = $this->lastResponse->getHeaderLine('Content-Type');
+            $errorMsg = $be->getMessage();
+            if (preg_match('!^application/json!', $contentType) || preg_match('!^text/plain!', $contentType)) {
+                // possibly content is a json-string containing a SolrException.
+                $solrException = \GuzzleHttp\json_decode($content);
+                if (property_exists($solrException, 'error') && property_exists($solrException->error, 'msg')) {
+                    $errorMsg = $solrException->error->msg;
+                }
             }
             if (defined('STDERR')) {
-                // fwrite(STDERR, $e->getRequest()->getBody()->getContents());
-                // fwrite(STDERR, "\n\n");
-                fwrite(STDERR, $e->getResponse()->getBody()->getContents());
+                fwrite(STDERR, PHP_EOL . $errorMsg . PHP_EOL);
+                fwrite(STDERR, PHP_EOL . $content . PHP_EOL);
             }
+            throw new Exception($errorMsg, null, $be);
```

Comment:
if the errorMsg holds a solr error message i think it would be better to have a custom exception instead of just a exception.

Reply:
This is already the custom exception coming from this bundle: src/Zicht/Bundle/SolrBundle/Solr/Exception.php